services/QuillLMS/lib/tasks/subscriptions.rake

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+namespace :subscriptions do
+  desc 'Add credit to users from a CSV file'
+  task :add_credit_to_users, [:filepath, :amount_in_days] => :environment do |t, args|
+    iostream = File.read(args[:filepath])
+    if (CSV.parse(iostream, headers: true).headers & ['email']).count != 1
+      puts 'Invalid headers. Exiting.'
+      exit 1
+    end
+
+    CSV.parse(iostream, headers: true) do |row|
+      email = row['email']
+      user = User.find_by_email(email)
+      if user.nil? || user.role != 'teacher'
+        puts "Unable to locate teacher with email #{email}"
+        next
+      end
+
+      begin
+        CreditTransaction.create!(user_id: user.id, amount: args[:amount_in_days].to_i, source: user)
```

Comment:
Not sure if there's an easy way to resolve this, but this line probably makes this script non-idempotent.  If it crashes halfway through, you can't just run it again with the same input file unless it's safe (and intended) to double-credit some users.

Reply:
Right - it's hard to make a script involving `CreditTransaction` idempotent. My solution was to log liberally, so that in the worst case we can create additional `CreditTransactions` on failed iterations. However, the script did not fail on staging, so I'm not expecting this to happen.

Reply:
You could create an ids array of created CreditTransactions and return them at the end of the rake task.  Then if you had to re-run the rake task cross check for user ids and remove those in the csv of already created credit transactions.

Reply:
I'm not sure about the `source: user` assignment in the CreditTransaction creation:  https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/app/models/credit_transaction.rb#L25

The code as-is seems to be creating a new source type without a corresponding action in the model file.  Not sure this matters but figured I would point it out.

Reply:
I added the transaction id logging. 

Regarding `source: user` - I believe it's the most logical thing, since the premium credit originates from the user's status as a TAC; they were not referred, and there is no pre-existing subscription. Also, we don't do much with this value anyway.